lib/Model/Message.php

```diff
@@ -161,6 +161,13 @@ public function isReplyable(): bool {
 			\in_array($this->getActorType(), [Attendee::ACTOR_USERS, Attendee::ACTOR_GUESTS]);
 	}
 
+	/**
+	 * Only my messages within 15 minutes after sending can be deleted
+	 */
+	public function isDeleteable(): bool {
+		return true;
+	}
```

Comment:
So for now we think this is not needed on the API, as the expression is rather constant:
```
message time > time() - 6h && (author === current user || current user is moderator)
```
Also we will check this on the deletion endpoint and therefor the time state needs to be updated live, as when you open the chat in the morning and have it open all day the time might expire, but your browser would not refetch the state (same for the mobile clients anyway).